### ISSUE TITLE
need to update the analyzer version

### ISSUE THREAD

**Issue by nikitadol** [dart-code-checker/dart-code-metrics#407]
```console
Because dart_code_metrics >=4.0.0-dev.1 depends on analyzer ^1.7.0 and json_serializable >=5.0.0 depends on analyzer ^2.0.0, dart_code_metrics >=4.0.0-dev.1 is incompatible with json_serializable >=5.0.0.
```

**Comment by incendial** [comment]
@nikitadol Hi, thanks for reporting the issue, right now the migration is blocked by the `analyzer_plugin` package we depend on. We'll update our package right after the Dart team releases the new version of it.

**Comment by nikitadol** [comment]
@incendial 
`analyzer_plugin: ^0.7.0` Already available

**Comment by incendial** [comment]
@nikitadol support for 2.0.0 analyzer is now available as 4.2.0-dev.2. We plan to release it as stable in a week or so.

**Comment by incendial** [comment]
Available in 4.2.0 release.
